shell/src/shell.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include "shell/src/streams.h"
+#include "shell/src/parser.h"
+#include "shell/src/commands/command.h"
+
+#include <string>
+
+namespace shell {
+
+class Shell {
+public:
+    Shell(const std::shared_ptr<IStreams>& streams) : streams_(streams) {}
+
+    void start();
+
+private:
+    enum class ShellStatus {
+        IDLE, RUNNING, FINISHED
+    };
+
+    void stop() { status_ = ShellStatus::FINISHED; }
+    bool is_running() const {return status_ == ShellStatus::RUNNING; }
+
+private:
+    std::shared_ptr<IStreams> streams_;
+    ShellStatus status_{ShellStatus::IDLE};
```

Comment:
General considerations.
In OS we have compiled binaries (single instances).  But they have state while running.

Certain command realization (echo etc) can have state inside private variables, or method execute can have its local vars.
From the outside the behavior is same, but I assume we should decide which way to implement.
Singleton command + local state inside `execute` 
or
Command class will have private vars inside (like here in shell now)

First variant looks more naturally (like in OS), but second more relates to OOP (Command object has state variables)

What do you think?